### PR TITLE
add mdx support

### PR DESCRIPTION
**Pull request by michaelglass**
adding mdx support for easier documentation writing (and proper markdown syntax highlighting)

quick side-track while writing voice compass docs

### CODE REVIEW

File: packages/website/src/components/PageContent.tsx [hunk 1]
```diff
@@ -40,17 +41,21 @@ export const Prose: FC<{ children: ReactNode; className?: string }> = ({
   </div>
 );
 
-export const PageContent: FC<{ md: string }> = ({ md }) => (
```
Comment (michaelglass) [inline packages/website/src/components/PageContent.tsx:43]: this is backwards compatible until / if we migrate documentation to support mdx

File: packages/website/package.json [hunk 2]
```diff
@@ -6,7 +6,10 @@
     "dev": "vite --force",
     "build": "tsc && vite build"
   },
+  "type": "module",
   "dependencies": {
```
Comment (michaelglass) [inline packages/website/package.json:10]: mdxjs plays nicer with ESM

File: packages/website/src/content/01-01-getting-started.mdx [hunk 3]
```diff
@@ -3,20 +3,14 @@ import { PageTitle } from "../components/PageTitle";
 import { PageContent } from "../components/PageContent";
 import { packageUrls } from "../constants";
 
-export const content = `
+
+<PageTitle pretitle="Introduction" title="Getting started" />
```
Comment (michaelglass) [inline packages/website/src/content/01-01-getting-started.mdx:7]: example mdx translation

File: packages/website/src/routes.tsx [hunk 4]
```diff
@@ -160,7 +160,7 @@ export const routes: {
       {
         label: "Getting started",
         url: "/headless-getting-started",
-        element: <HeadlessSetup />,
+        element: <HeadlessGettingStarted />,
```
Comment (michaelglass) [inline packages/website/src/routes.tsx:163]: unrelated cleanup I did along the way

**Review by jakub-nlx — APPROVED** [review]

**Review by peterszerzo — APPROVED** [review]

**Comment by michaelglass** [comment]
this seems to be a bad fit considering mdx [doesn't have native interpolation in code blocks](https://github.com/facebook/docusaurus/issues/5700)

the workarounds make dealing with code / pre decoration annoying. So stick with the status quo
